apps/backend/src/modules/projects/inactive-projects-notifier.ts

```diff
@@ -0,0 +1,105 @@
+import { EMPTY_LENGTH } from "~/libs/constants/constants.js";
+import { getDifferenceInDays } from "~/libs/helpers/helpers.js";
+import { type Logger } from "~/libs/modules/logger/logger.js";
+import { type NotificationService } from "~/modules/notifications/notification.service.js";
+import { type ProjectService } from "~/modules/projects/project.service.js";
+import { type UserService } from "~/modules/users/user.service.js";
+
+import {
+	NotificationDayThreshold,
+	NotificationProjectPermissionKey,
+	NotificationRootPermissionKey,
+} from "./libs/enums/enums.js";
+
+type Constructor = {
+	logger: Logger;
+	notificationService: NotificationService;
+	projectService: ProjectService;
+	userService: UserService;
+};
+
+class InactiveProjectsNotifier {
+	private logger: Logger;
+	private notificationService: NotificationService;
+	private projectService: ProjectService;
+	private userService: UserService;
+
+	public constructor({
+		logger,
+		notificationService,
+		projectService,
+		userService,
+	}: Constructor) {
+		this.projectService = projectService;
+		this.notificationService = notificationService;
+		this.userService = userService;
+		this.logger = logger;
+	}
+
+	private calculateInactiveDays(lastActivityDate: string): number {
+		const lastActivity = new Date(lastActivityDate);
+		const today = new Date();
+
+		return getDifferenceInDays(today, lastActivity);
+	}
+
+	public async processInactiveProjects(): Promise<void> {
+		const projects = await this.projectService.findInactiveProjects(
+			NotificationDayThreshold.FIRST_THRESHOLD,
+		);
+
+		for (const project of projects) {
+			if (!project.lastActivityDate) {
+				continue;
+			}
+
+			const daysInactive = this.calculateInactiveDays(project.lastActivityDate);
+
+			if (
+				daysInactive === NotificationDayThreshold.FIRST_THRESHOLD ||
+				daysInactive % NotificationDayThreshold.SECOND_THRESHOLD ===
+					EMPTY_LENGTH
```

Comment:
Let's store NotificationDayThreshold as array
```suggestion
const [firstThreshold, secondThreshold] = NOTIFICATION_DAY_THRESHOLD

			if (
				daysInactive === firstThreshold ||
				daysInactive % secondThreshold ===
					EMPTY_LENGTH
```